tests/subsys/rtio/rtio_api/src/rtio_iodev_test.h

```diff
@@ -55,7 +56,7 @@ static void rtio_iodev_test_next(struct rtio_iodev *iodev)
 static void rtio_iodev_timer_fn(struct k_timer *tm)
 {
 	struct rtio_iodev_test_data *data = CONTAINER_OF(tm, struct rtio_iodev_test_data, timer);
-	struct rtio_iodev_sqe *iodev_sqe = data->iodev_sqe;
+	struct rtio_iodev_sqe *iodev_sqe = (struct rtio_iodev_sqe *)data->iodev_sqe;
```

Comment:
The changes in the commit `rtio: Make api tests C++ compatible` should not be necessary since this test is not being compiled in the C++ mode anymore.

Reply:
Yeah, I was thinking of removing it, but the work was already done and if we ever want to change the test to .cpp it's ready.

Reply:
Done